src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -873,8 +873,11 @@ private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<TypeSy
                     diagnosticsBuilder.Add(new TypeParameterDiagnosticInfo(typeParameter, new CSDiagnosticInfo(ErrorCode.ERR_UnmanagedConstraintNotSatisfied, containingSymbol.ConstructedFrom(), typeParameter, typeArgument.Type)));
                     return false;
                 }
-                else if (managedKind == ManagedKind.UnmanagedWithGenerics)
+                else if (managedKind == ManagedKind.UnmanagedWithGenerics && currentCompilation is object)
                 {
+                    // In batch scenarios currentCompilation should always be non-null, but the public API
+                    // can check constraints in multiple places and there may not be a current compilation.
+                    // For these scenarios, we'll skip checking
```

Comment:
Is there any way that we can assert that if `currentCompilation` is null, we're not in a batch compilation scenario? Or is that a guaranteed fact in the first place?

Reply:
I don't know of a way to assert that, unfortunately.